### ISSUE TITLE
[Feature]: vibe_check_mentor - Senior engineer collaborative reasoning tool with Claude 4 Opus

### ISSUE THREAD

**Issue by kesslerio** [kesslerio/vibe-check-mcp#126]
## Problem Statement

Developers often need real-time feedback and guidance when making technical decisions, similar to bouncing ideas off a senior engineer. Current vibe-check tools provide static analysis and coaching, but lack the interactive, conversational aspect of collaborative problem-solving that happens naturally in pair programming or technical discussions.

## Proposed Solution

Create a new MCP tool called `vibe_check_mentor` that combines vibe-check pattern detection with clear-thought's collaborative reasoning capabilities, defaulting to Claude 4 Opus for superior reasoning performance.

### Core Features

1. **Interactive Conversational Interface**: Accept natural language input about technical decisions, architecture choices, or implementation approaches
2. **Automatic Pattern Detection**: Run vibe-check analysis on the input to detect potential anti-patterns
3. **Multi-Persona Collaborative Reasoning**: When issues are detected, automatically spawn a collaborative reasoning session with diverse engineering perspectives
4. **Claude 4 Opus Default**: Leverage Opus's superior reasoning for nuanced technical discussions
5. **Context-Aware Feedback**: Maintain conversation context for deeper, more relevant guidance

### Implementation Architecture

```python
@server.tool(
    name="vibe_check_mentor",
    description="Get senior engineer feedback on technical decisions through collaborative reasoning"
)
async def vibe_check_mentor(
    query: str,
    context: Optional[str] = None,
    session_id: Optional[str] = None,
    model: str = "opus",  # Default to Opus for best reasoning
    reasoning_depth: str = "comprehensive"  # quick | standard | comprehensive
) -> Dict[str, Any]:
    """
    Interactive senior engineer mentor combining vibe-check with collaborative reasoning.
    
    Args:
        query: Technical question or decision to discuss
        context: Additional context (code, architecture, requirements)
        session_id: Continue previous conversation
        model: Claude model to use (defaults to opus)
        reasoning_depth: Level of analysis depth
        
    Returns:
        Structured feedback with immediate insights and collaborative analysis
    """
```

### Reasoning Personas

The tool will utilize three specialized personas for comprehensive feedback:

1. **Senior Software Engineer**
   - Expertise: Architecture, best practices, technical debt
   - Perspective: Maintainability, scalability, proven solutions
   - Communication: Direct, practical guidance

2. **Product Engineer**
   - Expertise: MVP development, user value, rapid iteration
   - Perspective: Ship fast, get feedback, iterate
   - Communication: Pragmatic, user-focused

3. **AI/ML Engineer**
   - Expertise: AI integration, prompt engineering, tool design
   - Perspective: Human-AI collaboration, capability limits
   - Communication: Analytical, precise

### Workflow

1. **Input Processing**: Parse user query and context
2. **Initial Analysis**: Run vibe-check pattern detection
3. **Collaborative Reasoning** (if patterns detected):
   ```python
   # Spawn clear-thought collaborative reasoning
   reasoning_session = await spawn_collaborative_reasoning(
       topic=query,
       personas=ENGINEERING_PERSONAS,
       initial_analysis=vibe_check_results
   )
   ```
4. **Claude Code Integration**:
   ```python
   # Use Claude Code CLI with Opus
   cmd = ["claude", "--model", model, "-p", enhanced_prompt, "--output-format", "json"]
   response = await run_claude_cli(cmd)
   ```
5. **Response Synthesis**: Combine immediate feedback with deeper insights

### Example Usage

```python
# Quick architecture decision
result = await vibe_check_mentor(
    query="Should I build a custom HTTP client for this API integration?",
    context="Working with Stripe API, need webhook handling"
)

# Complex technical discussion
result = await vibe_check_mentor(
    query="Planning to implement event sourcing for our order system",
    context="Current monolith, 100k daily orders, team of 5 devs",
    reasoning_depth="comprehensive"
)

# Continue previous discussion
result = await vibe_check_mentor(
    query="What about using Kafka for the event bus?",
    session_id="previous-session-id"
)
```

### Expected Output Structure

```json
{
    "status": "success",
    "immediate_feedback": {
        "summary": "Consider using Stripe's official SDK instead",
        "confidence": 0.92,
        "detected_patterns": ["infrastructure_without_implementation"],
        "risk_level": "medium"
    },
    "collaborative_insights": {
        "consensus": ["Use official SDK", "Focus on business logic"],
        "perspectives": {
            "senior_engineer": "Official SDKs handle edge cases...",
            "product_engineer": "Ship faster with proven solutions...",
            "ai_engineer": "Modern SDKs include retry logic..."
        },
        "disagreements": [],
        "action_items": ["1. Install stripe-python", "2. Follow quickstart"]
    },
    "session_id": "mentor-session-xyz",
    "model_used": "claude-opus-4-20250514"
}
```

## Benefits

1. **Real-time Guidance**: Get senior engineer feedback without interrupting colleagues
2. **Multi-perspective Analysis**: Benefit from diverse engineering viewpoints
3. **Pattern Prevention**: Catch anti-patterns before they become technical debt
4. **Learning Tool**: Understand the "why" behind recommendations
5. **Context Retention**: Build on previous discussions for deeper insights

## Technical Requirements

- Integration with clear-thought MCP server for collaborative reasoning
- Claude Code CLI with model switching capability (see issue #125)
- Session management for conversation continuity
- Structured prompt engineering for optimal Opus performance

## Testing Plan

1. Unit tests for pattern detection integration
2. Mock collaborative reasoning sessions
3. Integration tests with Claude Code CLI
4. Real-world scenario testing with common decisions
5. Performance benchmarks (Opus vs Sonnet comparison)

## Future Enhancements

- Visual reasoning for architecture diagrams
- Code review mode for PR feedback
- Team knowledge integration (learn from CLAUDE.md)
- Slack/Discord integration for team discussions

## Related Issues

- Depends on #125 (model parameter support)
- Enhances existing vibe-check coaching framework
- Integrates with clear-thought collaborative reasoning

**Comment by kesslerio** [comment]
## 🔍 Updated Implementation Proposal: Borrowing from Clear-Thought

After reviewing the actual Clear-Thought server implementation, I've identified key patterns we can adapt for `vibe_check_mentor`. Their approach is actually quite simple - it's primarily a **data orchestration tool** that manages the collaborative reasoning state rather than generating the actual persona responses.

### 📚 Key Insights from Clear-Thought Implementation

**What Clear-Thought Actually Does:**
1. **State Management**: Tracks personas, contributions, consensus, disagreements
2. **Iteration Control**: Manages stages (problem-definition → ideation → critique → integration → decision → reflection)
3. **Output Formatting**: Pretty-prints the collaborative session with color coding
4. **Validation**: Ensures proper data structure for each iteration

**What It Doesn't Do:**
- No actual AI reasoning generation (expects external tool to provide contributions)
- No built-in persona logic or response generation
- Acts as a structured container for collaborative reasoning data

### 💡 Enhanced Implementation for vibe_check_mentor

We can create a **hybrid approach** that combines Clear-Thought's excellent data structures with our own persona reasoning engine:

```python
# src/vibe_check/tools/vibe_mentor.py

from typing import Dict, Any, List, Optional, Literal, Union
from dataclasses import dataclass, field
from enum import Enum
import json
import logging
from datetime import datetime

from ..core.vibe_coaching import VibeCoachingFramework, CoachingTone
from ..core.pattern_detector import PatternDetector
from ..tools.analyze_text_nollm import analyze_text_demo

logger = logging.getLogger(__name__)

# Borrowing Clear-Thought's data structures (adapted to Python)
@dataclass
class PersonaData:
    """Engineering persona definition - borrowed from Clear-Thought"""
    id: str
    name: str
    expertise: List[str]
    background: str
    perspective: str
    biases: List[str]
    communication: Dict[str, str]  # {"style": "...", "tone": "..."}

@dataclass
class ContributionData:
    """Single contribution in collaborative reasoning"""
    persona_id: str
    content: str
    type: Literal["observation", "question", "insight", "concern", "suggestion", "challenge", "synthesis"]
    confidence: float
    reference_ids: List[str] = field(default_factory=list)
    timestamp: str = field(default_factory=lambda: datetime.now().isoformat())

@dataclass
class DisagreementData:
    """Structured disagreement between personas"""
    topic: str
    positions: List[Dict[str, Any]]  # [{"personaId": "...", "position": "...", "arguments": [...]}]

@dataclass
class CollaborativeReasoningSession:
    """Complete session state - inspired by Clear-Thought"""
    topic: str
    personas: List[PersonaData]
    contributions: List[ContributionData]
    stage: Literal["problem-definition", "ideation", "critique", "integration", "decision", "reflection"]
    active_persona_id: str
    session_id: str
    iteration: int
    consensus_points: List[str] = field(default_factory=list)
    disagreements: List[DisagreementData] = field(default_factory=list)
    key_insights: List[str] = field(default_factory=list)
    open_questions: List[str] = field(default_factory=list)
    final_recommendation: Optional[str] = None
    next_contribution_needed: bool = True
    suggested_contribution_types: List[str] = field(default_factory=list)

class VibeMentorEngine:
    """
    Native collaborative reasoning engine for vibe-check mentor.
    Combines Clear-Thought's data structures with vibe-check pattern detection.
    """
    
    # Engineering personas adapted for vibe-check context
    DEFAULT_PERSONAS = [
        PersonaData(
            id="senior_engineer",
            name="Senior Software Engineer",
            expertise=["Architecture", "Best practices", "Technical debt", "Maintainability"],
            background="15 years building scalable systems, seen countless anti-patterns",
            perspective="Maintainability and proven solutions over novel approaches",
            biases=["Prefers established patterns", "Risk-averse", "Documentation-focused"],
            communication={"style": "Direct", "tone": "Professional"}
        ),
        PersonaData(
            id="product_engineer",
            name="Product Engineer", 
            expertise=["MVP development", "User value", "Rapid iteration", "Feature delivery"],
            background="Startup experience, shipped 50+ features under tight deadlines",
            perspective="Ship fast, iterate based on feedback, perfect is the enemy of done",
            biases=["Favors speed over perfection", "User-focused", "Pragmatic"],
            communication={"style": "Pragmatic", "tone": "Energetic"}
        ),
        PersonaData(
            id="ai_engineer",
            name="AI/ML Engineer",
            expertise=["AI integration", "Prompt engineering", "MCP tools", "LLM patterns"],
            background="Deep learning research, extensive Claude and GPT integrations",
            perspective="Human-AI collaboration patterns, tool-augmented development",
            biases=["Excited about AI capabilities", "Pattern-focused", "Tool-first thinking"],
            communication={"style": "Analytical", "tone": "Precise"}
        )
    ]
    
    def __init__(self):
        self.sessions: Dict[str, CollaborativeReasoningSession] = {}
        self.pattern_detector = PatternDetector()
        
    def create_session(
        self, 
        topic: str,
        personas: Optional[List[PersonaData]] = None,
        session_id: Optional[str] = None
    ) -> CollaborativeReasoningSession:
        """Initialize a new collaborative reasoning session"""
        
        session_id = session_id or f"mentor-session-{hash(topic) % 100000}"
        
        session = CollaborativeReasoningSession(
            topic=topic,
            personas=personas or self.DEFAULT_PERSONAS,
            contributions=[],
            stage="problem-definition",
            active_persona_id=self.DEFAULT_PERSONAS[0].id,
            session_id=session_id,
            iteration=0,
            suggested_contribution_types=["observation", "question"]
        )
        
        self.sessions[session_id] = session
        return session
        
    def generate_contribution(
        self,
        session: CollaborativeReasoningSession,
        persona: PersonaData,
        detected_patterns: List[Dict[str, Any]],
        context: Optional[str] = None
    ) -> ContributionData:
        """
        Generate a contribution from a persona based on their characteristics.
        This is our enhancement over Clear-Thought - actual reasoning generation.
        """
        
        # Analyze context for persona-specific insights
        contribution_type, content, confidence = self._reason_as_persona(
            persona, 
            session.topic,
            detected_patterns,
            session.contributions,
            context
        )
        
        contribution = ContributionData(
            persona_id=persona.id,
            content=content,
            type=contribution_type,
            confidence=confidence,
            reference_ids=self._find_references(content, session.contributions)
        )
        
        return contribution
        
    def _reason_as_persona(
        self,
        persona: PersonaData,
        topic: str,
        patterns: List[Dict[str, Any]],
        previous_contributions: List[ContributionData],
        context: Optional[str]
    ) -> tuple[str, str, float]:
        """
        Simulate persona reasoning based on their expertise and biases.
        Returns: (contribution_type, content, confidence)
        """
        
        # Pattern-based reasoning for each persona
        if persona.id == "senior_engineer":
            # Senior engineer focuses on maintainability and established patterns
            if any(p["pattern_type"] == "infrastructure_without_implementation" for p in patterns):
                return (
                    "concern",
                    "This looks like infrastructure-first thinking. In my experience spanning 15 years, "
                    "we should always start with working API calls before building abstractions. "
                    "I strongly recommend following the official SDK examples first - they handle edge cases "
                    "we'll inevitably miss in custom implementations.",
                    0.92
                )
            elif "custom" in topic.lower() and any(word in topic.lower() for word in ["http", "client", "api"]):
                return (
                    "insight",
                    "Building custom HTTP clients is rarely necessary and often a maintenance burden. "
                    "Most services provide official SDKs that handle retry logic, authentication, rate limiting, "
                    "and error handling. Let's check for an official solution first - it could save weeks of work.",
                    0.88
                )
            else:
                return (
                    "suggestion",
                    "For long-term maintainability, I suggest starting with the simplest solution that works. "
                    "We can always add complexity later if truly needed. Focus on clear documentation, "
                    "standard patterns, and making it easy for the next developer to understand.",
                    0.85
                )
                
        elif persona.id == "product_engineer":
            # Product engineer focuses on shipping value quickly
            if patterns:
                return (
                    "suggestion",
                    "Let's not overthink this! What's the fastest way to deliver value to users? "
                    "I'd build a quick prototype with the official tools, get it in front of users this week, "
                    "and iterate based on real feedback. Remember, users don't care about our architecture - "
                    "they care about solving their problems.",
                    0.90
                )
            elif "planning" in topic.lower() or "architecture" in topic.lower():
                return (
                    "challenge",
                    "Are we solving a real user problem or just satisfying our engineering desires? "
                    "I've shipped 50+ features, and the ones that succeed focus on user value, not technical elegance. "
                    "Can we validate this with users before investing heavily in the implementation?",
                    0.87
                )
            else:
                return (
                    "observation",
                    "This sounds good from a product perspective! Can we ship something basic this week and iterate? "
                    "In my startup experience, the first version is never perfect - but it teaches us what users actually need.",
                    0.85
                )
                
        elif persona.id == "ai_engineer":
            # AI engineer focuses on modern tools and AI assistance
            if "integration" in topic.lower() or "ai" in topic.lower():
                return (
                    "insight",
                    "Modern AI services provide excellent SDKs that handle the complexity for us. "
                    "For example, Claude's SDK manages streaming, tool use, context windows, and retry logic. "
                    "Custom implementations often miss critical edge cases like token limits, rate limiting, "
                    "and proper error handling that the official SDK handles elegantly.",
                    0.89
                )
            elif previous_contributions:
                # Synthesize previous contributions
                return (
                    "synthesis",
                    "Building on the previous points, I see a pattern here: we're considering building "
                    "infrastructure before proving the basic functionality. Modern AI tools can accelerate "
                    "our development - MCP tools, GitHub Copilot, and structured prompts can help generate "
                    "boilerplate and catch anti-patterns early. Let's leverage these tools.",
                    0.86
                )
            else:
                return (
                    "suggestion",
                    "Consider how AI tools can accelerate this work. MCP tools can provide immediate feedback, "
                    "GitHub Copilot can generate boilerplate, and LLMs can help validate our approach. "
                    "Why build from scratch when AI can help us prototype 10x faster?",
                    0.82
                )
        
        # Default response
        return (
            "observation",
            f"From my {persona.name} perspective with expertise in {', '.join(persona.expertise[:2])}, "
            f"this requires careful consideration of trade-offs.",
            0.70
        )
        
    def advance_stage(self, session: CollaborativeReasoningSession) -> str:
        """Advance to the next stage in the reasoning process"""
        
        stage_progression = {
            "problem-definition": "ideation",
            "ideation": "critique",
            "critique": "integration", 
            "integration": "decision",
            "decision": "reflection",
            "reflection": "reflection"  # Terminal stage
        }
        
        session.stage = stage_progression.get(session.stage, "reflection")
        session.iteration += 1
        
        # Update suggested contribution types based on stage
        stage_suggestions = {
            "problem-definition": ["observation", "question"],
            "ideation": ["suggestion", "insight"],
            "critique": ["concern", "challenge"],
            "integration": ["synthesis", "insight"],
            "decision": ["synthesis", "suggestion"],
            "reflection": ["observation", "insight"]
        }
        
        session.suggested_contribution_types = stage_suggestions.get(
            session.stage, 
            ["observation"]
        )
        
        return session.stage
        
    def synthesize_session(self, session: CollaborativeReasoningSession) -> Dict[str, Any]:
        """
        Synthesize the collaborative reasoning session into actionable insights.
        This is our key value-add over Clear-Thought's simple state tracking.
        """
        
        # Group contributions by type
        contributions_by_type = {}
        for contrib in session.contributions:
            if contrib.type not in contributions_by_type:
                contributions_by_type[contrib.type] = []
            contributions_by_type[contrib.type].append(contrib)
        
        # Extract consensus points (similar content from multiple personas)
        consensus = []
        all_content = [c.content.lower() for c in session.contributions]
        
        # Simple consensus detection based on keyword overlap
        consensus_keywords = ["official", "sdk", "simple", "prototype", "user", "feedback"]
        for keyword in consensus_keywords:
            if sum(1 for content in all_content if keyword in content) >= 2:
                consensus.append(f"Use {keyword} approaches when available")
        
        # Identify key insights (high confidence insights and syntheses)
        key_insights = []
        for contrib in session.contributions:
            if contrib.type in ["insight", "synthesis"] and contrib.confidence > 0.85:
                key_insights.append(contrib.content)
        
        # Find disagreements
        disagreements = []
        concerns = contributions_by_type.get("concern", [])
        challenges = contributions_by_type.get("challenge", [])
        
        if concerns or challenges:
            for item in concerns + challenges:
                # Check if there's a counter-position
                disagreements.append({
                    "topic": "Implementation approach",
                    "positions": [
                        {
                            "personaId": item.persona_id,
                            "position": item.content,
                            "arguments": [item.content]
                        }
                    ]
                })
        
        # Generate final recommendation
        if session.stage in ["decision", "reflection"]:
            # Prioritize synthesis contributions
            syntheses = contributions_by_type.get("synthesis", [])
            if syntheses:
                final_rec = syntheses[-1].content
            else:
                final_rec = "Based on the discussion, start with the simplest official solution and iterate."
            session.final_recommendation = final_rec
        
        # Build comprehensive summary
        return {
            "session_summary": {
                "topic": session.topic,
                "stage": session.stage,
                "iterations": session.iteration,
                "total_contributions": len(session.contributions)
            },
            "consensus_points": consensus,
            "key_insights": key_insights[:3],  # Top 3
            "primary_concerns": [c.content for c in concerns[:2]],
            "disagreements": disagreements,
            "recommendations": {
                "immediate_actions": [
                    "Research official SDK/documentation",
                    "Create minimal proof of concept",
                    "Validate with real data",
                    "Get early user feedback"
                ],
                "avoid": [
                    "Building custom infrastructure first",
                    "Over-engineering the solution",
                    "Skipping official documentation",
                    "Making assumptions without validation"
                ]
            },
            "final_recommendation": session.final_recommendation
        }
        
    def format_session_output(self, session: CollaborativeReasoningSession) -> str:
        """
        Format session for display - inspired by Clear-Thought's chalk formatting.
        Using ANSI codes for terminal colors.
        """
        
        BLUE = "\033[34m"
        GREEN = "\033[32m"
        YELLOW = "\033[33m"
        MAGENTA = "\033[35m"
        CYAN = "\033[36m"
        BOLD = "\033[1m"
        RESET = "\033[0m"
        
        output = f"\n{BOLD}{BLUE}🧠 Collaborative Reasoning Session{RESET}\n"
        output += f"{BOLD}{GREEN}Topic:{RESET} {session.topic}\n"
        output += f"{BOLD}{YELLOW}Stage:{RESET} {session.stage} (Iteration: {session.iteration})\n"
        
        # Active persona
        active_persona = next((p for p in session.personas if p.id == session.active_persona_id), None)
        if active_persona:
            output += f"\n{BOLD}{MAGENTA}Active Persona:{RESET} {active_persona.name}\n"
            output += f"{BOLD}{CYAN}Expertise:{RESET} {', '.join(active_persona.expertise)}\n"
            output += f"{BOLD}{CYAN}Perspective:{RESET} {active_persona.perspective}\n"
        
        # Contributions
        if session.contributions:
            output += f"\n{BOLD}{GREEN}Contributions:{RESET}\n"
            for contrib in session.contributions:
                persona = next((p for p in session.personas if p.id == contrib.persona_id), None)
                persona_name = persona.name if persona else contrib.persona_id
                
                output += f"\n{BOLD}{persona_name} ({contrib.type}, confidence: {contrib.confidence:.2f}):{RESET}\n"
                output += f"{contrib.content}\n"
        
        # Consensus points
        if session.consensus_points:
            output += f"\n{BOLD}{GREEN}Consensus Points:{RESET}\n"
            for i, point in enumerate(session.consensus_points, 1):
                output += f"{BOLD}{i}.{RESET} {point}\n"
        
        # Key insights
        if session.key_insights:
            output += f"\n{BOLD}{YELLOW}Key Insights:{RESET}\n"
            for i, insight in enumerate(session.key_insights, 1):
                output += f"{BOLD}{i}.{RESET} {insight}\n"
        
        # Final recommendation
        if session.final_recommendation:
            output += f"\n{BOLD}{CYAN}Final Recommendation:{RESET}\n"
            output += f"{session.final_recommendation}\n"
        
        return output
        
    def _find_references(self, content: str, contributions: List[ContributionData]) -> List[str]:
        """Find contributions that this content references"""
        references = []
        content_lower = content.lower()
        
        for contrib in contributions:
            # Simple reference detection based on keyword overlap
            if any(word in content_lower for word in contrib.content.lower().split()[:5]):
                references.append(f"{contrib.persona_id}_{contrib.type}")
        
        return references


# FastMCP tool implementation
@server.tool(
    name="vibe_check_mentor",
    description="Get senior engineer feedback through collaborative reasoning - multiple engineering perspectives analyze your technical decisions"
)
async def vibe_check_mentor(
    query: str,
    context: Optional[str] = None,
    session_id: Optional[str] = None,
    reasoning_depth: str = "standard",  # quick | standard | comprehensive
    continue_session: bool = False
) -> Dict[str, Any]:
    """
    Interactive senior engineer mentor combining vibe-check with collaborative reasoning.
    Inspired by Clear-Thought's collaborative reasoning with native implementation.
    """
    
    logger.info(f"Vibe mentor activated: {reasoning_depth} analysis")
    
    try:
        # Initialize engine
        engine = VibeMentorEngine()
        
        # Step 1: Run vibe-check pattern detection
        combined_text = f"{query}\n\n{context}" if context else query
        vibe_analysis = analyze_text_demo(combined_text, detail_level="standard")
        
        detected_patterns = vibe_analysis.get("detected_patterns", [])
        vibe_level = vibe_analysis.get("vibe_assessment", {}).get("vibe_level", "unknown")
        
        # Step 2: Create or retrieve session
        if continue_session and session_id and session_id in engine.sessions:
            session = engine.sessions[session_id]
            # Update topic for continued conversation
            session.topic = query
        else:
            session = engine.create_session(
                topic=query,
                session_id=session_id
            )
        
        # Step 3: Determine number of contributions based on depth
        contribution_counts = {
            "quick": 1,  # Just senior engineer
            "standard": 2,  # Senior + Product  
            "comprehensive": 3  # All personas
        }
        
        num_contributions = contribution_counts.get(reasoning_depth, 2)
        
        # Step 4: Generate contributions from personas
        for i in range(num_contributions):
            if i < len(session.personas):
                persona = session.personas[i]
                session.active_persona_id = persona.id
                
                contribution = engine.generate_contribution(
                    session=session,
                    persona=persona,
                    detected_patterns=detected_patterns,
                    context=context
                )
                
                session.contributions.append(contribution)
                
                # Advance stage after each contribution in comprehensive mode
                if reasoning_depth == "comprehensive" and i < num_contributions - 1:
                    engine.advance_stage(session)
        
        # Step 5: Synthesize insights
        synthesis = engine.synthesize_session(session)
        
        # Step 6: Get coaching recommendations
        coaching_framework = VibeCoachingFramework()
        coaching_recs = coaching_framework.generate_coaching_recommendations(
            vibe_level=vibe_level,
            detected_patterns=[],  # Already processed
            issue_context={"query": query},
            tone=CoachingTone.ENCOURAGING
        )
        
        # Step 7: Build response
        response = {
            "status": "success",
            "immediate_feedback": {
                "summary": _generate_summary(vibe_level, detected_patterns),
                "confidence": vibe_analysis.get("vibe_assessment", {}).get("confidence", 0),
                "detected_patterns": [p["pattern_type"] for p in detected_patterns],
                "vibe_level": vibe_level
            },
            "collaborative_insights": {
                "consensus": synthesis["consensus_points"],
                "perspectives": {
                    contrib.persona_id: {
                        "message": contrib.content,
                        "type": contrib.type,
                        "confidence": contrib.confidence
                    }
                    for contrib in session.contributions
                },
                "key_insights": synthesis["key_insights"],
                "concerns": synthesis["primary_concerns"],
                "recommendations": synthesis["recommendations"]
            },
            "coaching_guidance": {
                "primary_recommendation": coaching_recs[0].title if coaching_recs else "Proceed with implementation",
                "action_steps": coaching_recs[0].action_items[:3] if coaching_recs else [],
                "prevention_checklist": coaching_recs[0].prevention_checklist[:3] if coaching_recs else []
            },
            "session_info": {
                "session_id": session.session_id,
                "stage": session.stage,
                "iteration": session.iteration,
                "can_continue": session.next_contribution_needed
            },
            "reasoning_depth": reasoning_depth,
            "formatted_output": engine.format_session_output(session)
        }
        
        # Log formatted output for debugging
        logger.info(response["formatted_output"])
        
        return response
        
    except Exception as e:
        logger.error(f"Vibe mentor error: {e}", exc_info=True)
        return {
            "status": "error",
            "message": f"Mentoring session failed: {str(e)}",
            "fallback_guidance": [
                "Start with official documentation",
                "Build a simple prototype first",
                "Get feedback early and often"
            ]
        }
```

### 🎯 Key Improvements from Clear-Thought Study

1. **Proper Data Structures**: Using their well-designed interfaces for personas, contributions, and sessions
2. **Stage Management**: Implementing their stage progression (problem-definition → ideation → critique → integration → decision → reflection)
3. **Contribution Types**: Supporting all 7 types (observation, question, insight, concern, suggestion, challenge, synthesis)
4. **Session Continuity**: Adding session management for multi-turn conversations
5. **Formatted Output**: Color-coded terminal output inspired by their chalk formatting

### 🚀 Additional Enhancements

1. **Native Reasoning Engine**: Unlike Clear-Thought which just manages state, we generate actual persona responses
2. **Vibe-Check Integration**: Pattern detection drives persona responses
3. **Synthesis Logic**: Automated consensus detection and insight extraction
4. **Coaching Integration**: Combines with existing vibe coaching framework

### 📊 Example Usage with Clear-Thought Patterns

```python
# Initial query
result = await vibe_check_mentor(
    query="Should I build a custom auth system for our app?",
    reasoning_depth="comprehensive"
)

# Continue session with follow-up
result = await vibe_check_mentor(
    query="What about using Supabase Auth instead?",
    session_id=result["session_info"]["session_id"],
    continue_session=True,
    reasoning_depth="standard"
)
```

### 🔄 Migration Path

1. **Phase 1**: Implement core with native reasoning (1 week)
2. **Phase 2**: Add Clear-Thought MCP integration for enhanced reasoning (optional)
3. **Phase 3**: Session persistence and advanced features

This implementation truly leverages the best of both worlds - Clear-Thought's excellent collaborative reasoning structure with vibe-check's pattern detection and educational focus!
